google/cloud/logging_v2/logger.py

```diff
@@ -162,11 +164,15 @@ def _do_log(self, client, _entry_class, payload=None, **kw):
 
         api_repr = entry.to_api_repr()
         entries = [api_repr]
+
         if google.cloud.logging_v2._instrumentation_emitted is False:
             entries = _add_instrumentation(entries, **kw)
             google.cloud.logging_v2._instrumentation_emitted = True
         # partial_success is true to avoid dropping instrumentation logs
-        client.logging_api.write_entries(entries, partial_success=True)
+        try:
+            client.logging_api.write_entries(entries, partial_success=True)
+        except Exception:
+            _LOGGER.error("Failed to submit log message.", exc_info=True)
```

Comment:
I don't think it's a good idea to just swallow all exceptions here. All grpc networking errors would now be ignored (ex bad IAM credentials), and users will have no idea anything is wrong unless they know to subscribe to the internal _LOGGER. And existing customers may have error handling logic in place that will now be bypassed, so this could be a breaking change hidden in a bug fix

Reply:
You swallowing catch the specific exception instead of catching all exceptions?

Honestly, I think just raising an exception here makes sense, so the user knows their input isn't handled. I'd do a `raise ValueError(...) from e` though, so we can give more context instead of raising the raw internal error. And then add a `Raises:` section in the docstring describing this situation